app/controllers/categories_controller.rb

```diff
@@ -0,0 +1,45 @@
+class CategoriesController < ApplicationController
```

Comment:
Under the path ``/categories/id`` there is `edit` link, which is visible for not logged in users. You might forgot to add some code to prevent not logged in users (like me) from editing categories.